packages/merge-styles/src/shadowConfig.ts

```diff
@@ -24,10 +26,14 @@ export const makeShadowConfig = (stylesheetKey: string, inShadow: boolean, windo
   };
 };
 
-export const isShadowConfig = (obj: unknown): obj is ShadowConfig => {
+export const isShadowConfig = <T extends ShadowConfig | PartialShadowConfig>(obj: unknown): obj is T => {
```

Comment:
this is a mistake actually from TS POV, it will return true only if the shape is actual `ShadowConfig` not partial. 

so   `const isShadowConfig = <T extends ShadowConfig | PartialShadowConfig>(obj: unknown): obj is ShadowConfig`

but that generic is not necessary so it can be 
`const isShadowConfig = (obj: unknown): obj is ShadowConfig`